Detectors/AOD/src/AODMcProducerWorkflowSpec.cxx

```diff
@@ -223,102 +194,66 @@ void AODMcProducerWorkflowDPL::init(InitContext& ic)
     mMcParticleMom = 0xFFFFFFFF;
   }
 
+  // parse list of sim prefixes into vector
+  mMCHeaderFNames = ic.options().get<std::string>("mckine-fnames");
+  std::stringstream ss(mMCHeaderFNames);
+  while (ss.good()) {
+    std::string substr;
+    std::getline(ss, substr, ',');
+    mSimPrefixes.push_back(substr);
+  }
+
   mTimer.Reset();
 }
 
 void AODMcProducerWorkflowDPL::run(ProcessingContext& pc)
 {
   mTimer.Start(false);
-  mStartIR = pc.services().get<o2::framework::TimingInfo>().firstTForbit;
 
-  const auto& tinfo = pc.services().get<o2::framework::TimingInfo>();
+  uint64_t tfNumber = mTFNumber;
 
-  auto& bcBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "BC"});
   auto& mcCollisionsBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "MCCOLLISION"});
   auto& mcParticlesBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "MCPARTICLE_001"});
   auto& originTableBuilder = pc.outputs().make<TableBuilder>(Output{"AOD", "ORIGIN"});
 
-  auto bcCursor = bcBuilder.cursor<o2::aod::BCs>();
   auto mcCollisionsCursor = mcCollisionsBuilder.cursor<o2::aod::McCollisions>();
   auto mcParticlesCursor = mcParticlesBuilder.cursor<o2::aod::StoredMcParticles_001>();
   auto originCursor = originTableBuilder.cursor<o2::aod::Origins>();
 
-  auto mcReader = std::make_unique<o2::steer::MCKinematicsReader>("collisioncontext.root");
```

Comment:
I think there was a use case when taking collisioncontext was justified: When signals are embedded in background events.
Is the new code still doing this correctly?

(If not I would offer 2 options: One with just kinematics and one where collisioncontext is available with the information of which events are embedded onto which other events).

Reply:
Hi @sawenzel

I have added an option to use collisioncontext to process embedded events properly. Could you please take a look?

Reply:
To be clear, I asked for an optional path in which the collision context isn't needed, but I think this should be an option - there is a use case in which this could be switched on. Thanks a lot, Nazar!!